src/transfer-manager.ts

```diff
@@ -667,24 +676,39 @@ export class TransferManager {
             end: chunkEnd,
             [GCCL_GCS_CMD_KEY]: GCCL_GCS_CMD_FEATURE.DOWNLOAD_SHARDED,
           });
-          return fileToWrite.write(resp[0], 0, resp[0].length, chunkStart);
+          const result = await fileToWrite.write(
+            resp[0],
+            0,
+            resp[0].length,
+            chunkStart
+          );
+          if (noReturnData) return;
+          return result.buffer;
         })
       );
 
       start += chunkSize;
     }
 
-    let results: DownloadResponse;
+    let chunks: Array<Buffer | void>;
     try {
-      const data = await Promise.all(promises);
-      results = data.map(result => result.buffer) as DownloadResponse;
-      if (options.validation === 'crc32c') {
-        await CRC32C.fromFile(filePath);
-      }
-      return results;
+      chunks = await Promise.all(promises);
```

Comment:
[Optional]: It might be cleaner to only set `chunks` if `!noReturnData` to avoid casting later, but no biggie.

Reply:
Going to leave that as is if that's OK, imo it's clear on the return with `if (!noReturnData)` (and not returning anything) rather than checking `if (chunks)` and then returning the `Buffer.concat`.